diff -urN john-1.6.orig/src/Makefile john-1.6/src/Makefile
--- john-1.6.orig/src/Makefile	Thu Dec  3 01:29:50 1998
+++ john-1.6/src/Makefile	Wed Dec  9 12:25:28 1998
@@ -116,13 +116,13 @@
 	$(LN) x86-any.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486"
+		CFLAGS="$(CFLAGS)"
 
 linux-x86-mmx-elf:
 	$(LN) x86-mmx.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486"
+		CFLAGS="$(CFLAGS)"
 
 linux-x86-k6-elf:
 	$(LN) x86-k6.h arch.h
@@ -133,7 +133,7 @@
 	$(LN) x86-any.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DUNDERSCORES -DALIGN_LOG"
 
 linux-alpha:
@@ -160,7 +160,7 @@
 	$(LN) x86-any.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DUNDERSCORES -DALIGN_LOG -DBSD"
 
 freebsd-x86-k6-a.out:
@@ -173,14 +173,14 @@
 	$(LN) x86-any.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DBSD"
 
 freebsd-x86-mmx-elf:
 	$(LN) x86-mmx.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DBSD"
 
 freebsd-x86-k6-elf:
@@ -193,7 +193,7 @@
 	$(LN) x86-any.h arch.h
 	$(MAKE) $(PROJ) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DUNDERSCORES -DALIGN_LOG -DBSD"
 
 openbsd-x86-k6:
@@ -260,7 +260,7 @@
 	$(MAKE) $(PROJ) \
 		SHELL=/bin/sh \
 		JOHN_OBJS="$(JOHN_OBJS) solaris-x86.o" \
-		CFLAGS="$(CFLAGS) -m486"
+		CFLAGS="$(CFLAGS)"
 
 solaris-x86-k6:
 	$(RM) arch.h
@@ -340,14 +340,14 @@
 	copy x86-any.h arch.h
 	$(MAKE) $(PROJ_DOS) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DUNDERSCORES -DALIGN_LOG"
 
 dos-djgpp-x86-mmx:
 	copy x86-mmx.h arch.h
 	$(MAKE) $(PROJ_DOS) \
 		JOHN_OBJS="$(JOHN_OBJS) x86.o" \
-		CFLAGS="$(CFLAGS) -m486" \
+		CFLAGS="$(CFLAGS)" \
 		ASFLAGS="$(ASFLAGS) -DUNDERSCORES -DALIGN_LOG"
 
 dos-djgpp-x86-k6:
