--- john-1.7.9/src/MD5_std.c~	2013-07-21 13:22:20.285649779 +0200
+++ john-1.7.9/src/MD5_std.c	2013-07-21 13:22:44.392541796 +0200
@@ -472,12 +472,14 @@
 /*
  * x86-64 implies a fairly recent CPU, so presumably its L1 instruction cache
  * is large enough.
- */
 #ifdef __x86_64__
 #define MAYBE_INLINE_BODY MAYBE_INLINE
 #else
+ */
 #define MAYBE_INLINE_BODY
+/*
 #endif
+ */
 
 #if !MD5_X2
 
--- john-1.7.9/src/lotus5_fmt_plug.c~	2013-07-21 13:25:04.637186087 +0200
+++ john-1.7.9/src/lotus5_fmt_plug.c	2013-07-21 13:25:41.664198855 +0200
@@ -155,7 +155,7 @@
 /*Beginning of private functions*/
 /* Takes the plaintext password and generates the second row of our
  * working matrix for the final call to the mixing function*/
-void MAYBE_INLINE
+void
 lotus_transform_password (unsigned char *i1, unsigned char *o1,
     unsigned char *i2, unsigned char *o2)
 {
@@ -173,7 +173,7 @@
 }
 
 /* The mixing function: perturbs the first three rows of the matrix*/
-void MAYBE_INLINE lotus_mix (unsigned char *m1, unsigned char *m2)
+void lotus_mix (unsigned char *m1, unsigned char *m2)
 {
   int i, j;
   unsigned char p1, p2;
